--- a/Makefile
+++ b/Makefile
@@ -12,11 +12,14 @@ version:		dkms.conf
 netatop.ko: 		netatop.c
 			$(MAKE) -C $(KERNDIR) M=$(THISDIR) modules

-netatopd:		netatopd.o Makefile
-			$(CC) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) netatopd.o -o netatopd -lz
+netatopd.o:		netatopd.c netatop.h netatopversion.h netatopd.h
+			$(HOSTCC) $(CPPFLAGS) $(CFLAGS) -c -o netatopd.o netatopd.c
+
+netatopd:		netatopd.o
+			$(HOSTCC) $(LDFLAGS) netatopd.o -o netatopd -lz

 netatop.ko:		netatop.h   netatopversion.h
-netatopd.o:		netatop.h   netatopversion.h   netatopd.h
+


 install:		netatop.ko netatopd
